pro/src/new_components/ButtonDownloadCSV/__specs__/ButtonDownloadCSV.spec.jsx

```diff
@@ -20,28 +20,28 @@ const renderButtonDownloadCSV = async ({ props, storeOverrides }) => {
 describe('src | components | Layout | ButtonDownloadCSV', () => {
   describe('render', () => {
     it('should disable button during download', async () => {
-      await new Promise(resolve => {
-        fetch.mockResponse(JSON.stringify({}), { status: 200 })
+      // await new Promise(resolve => {
+      fetch.mockResponse(JSON.stringify({}), { status: 200 })
 
-        const props = {
-          filename: 'test-csv',
-          href: 'http://test.com',
-          mimeType: 'text/csv',
-          isDisabled: false,
-        }
+      const props = {
+        filename: 'test-csv',
+        href: 'http://test.com',
+        mimeType: 'text/csv',
+        isDisabled: false,
+      }
 
-        renderButtonDownloadCSV({ props })
+      renderButtonDownloadCSV({ props })
 
-        const button = screen.getByText('Fake Title')
+      const button = await screen.findByText('Fake Title')
+      expect(button).toBeEnabled()
+
+      // maybe we need fireEvent here
+      fireEvent.click(button)
```

Comment:
ça ne fonctionne pas avec userEvent ?

Reply:
mais en gros en général c'est des tests où on veut expect sur un résultat intermédiaire, avec le userEvent j'ai l'impression d'être toujours après cet étatr intermédiaire

Reply:
ok, même si on met pas await devant le userEvent ?

Reply:
yep ! mais pareil si tu trouves mieux je veux bien changer :)

Reply:
ça semble fonctionner, en omettant le `await` devant le premier `waitFor`
```
 await userEvent.click(button)
      waitFor(() => {
        expect(button).toBeDisabled()
      })
      await waitFor(() => {
        expect(button).toBeEnabled()
      })
```

Reply:
je veux bien essayer mais c'est contre indiqué par la doc
`The async methods return a Promise, so you must always use await or .then(done) when calling them.`
https://testing-library.com/docs/guide-disappearance/

Reply:
c'est bien un faux positif ce test passe aussi !

```
      await userEvent.click(button)

      // then
      waitFor(() => {
        expect(screen.getByText('Fake qzmlfjqzmriofjTitle')).toBeInTheDocument()

        expect(button).toBeDisabled()
      })
      await waitFor(() => {
        expect(button).toBeEnabled()
      })
```

Reply:
et ici https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event
il dit que c'est pas toujours possible d'utiliser useEvent 
`Advice: Use @testing-library/user-event over fireEvent where possible.`

Reply:
c'est vrai mais ça nous fait une lib pour 3 tests dans la codebase, ce serait mieux d'arriver à écrire le test autrement - si on mocke la réponse en erreur, le bouton devrait rester en disabled et on pourrait le tester ?

Reply:
j'ai rajoué le test mais en fait il reste pas disabled en cas d'erreur (et je pense que c'est voulu)
donc ça ne fonctionne pas dans notre cas

Reply:
ok bon tant pis laissons comme ça, on fera un chantier pour se débarrasser d'userEvent